### DISCUSSION TITLE
Vs code server rendering

### DISCUSSION THREAD

**Discussion by GooseyBoy1000** [coder/code-server#4954] [Q&A]
Hello, I was wondering if you can renderer things on vs code when you run some code like if you wanted to make a c++ application could you create a window with a 3d model on it. If not is it hypothetically possible to do it yourself by rendering the image on the machine and then communicating to the server to output the desired thing.

**Comment by jsjoeio** [comment]
Hmm...I'm not sure? I haven't developed c++ applications before so I'm not familiar with the requirements and limitations. 

Maybe someone else from the community can chime in though.

**Reply by GooseyBoy1000** [reply]
I am not just talking about c++ code. What i am saying is when you run a .exe file you normaly get a window of some sort could it be possible to get that window and output everything in it in a server so you can view it with a browser?

**Reply by jsjoeio** [reply]
Hmm...I'm not sure I follow. You mean run code-server as a desktop application but pipe it to a server?

It might help to hear your end goal. What are you trying to acomplish?

**Reply by GooseyBoy1000** [reply]
What I am saying is when you run code you have the ability to render stuff in a window but in the browser.

**Reply by jsjoeio** [reply]
Do you mean run an application/server inside code-server and then open that application/server in a new browser window?

**Reply by lg-nineoneninethreenine** [reply]
I think he is asking for the code output. You can get it from the terminal if you run your code. Alternatively, you could set up VNC and possibly connect from that too.

**Reply by lg-nineoneninethreenine** [reply]
That might allow a GUI so you could see the output in a window.

**Reply by GooseyBoy1000** [reply]
Ok thanks

**Reply by GooseyBoy1000** [reply]
I think I am just going to code a way to screen record the right application and then output the video into a web server.
